### DISCUSSION TITLE
Z-Stack Router firmware and default power transmission

### DISCUSSION THREAD

**Discussion by hesham1216** [Koenkk/Z-Stack-firmware#521] [Q&A] [answered]
Hello Guys,

I am building a Zigbee router PCB and it will be based on CC2652P2 as the firmwares available are for those chips.

Just wanted to know what is the default power transmission of the default firmware, and can it be modifed somehow if it is less than the 20db?

I know and as said before it is not always about the power transmission, but the mesh network I am building is complicated and have more than +140 zigbee devices.

I was using my own designed PCB coordinator as well, and manufactured some pcb's and used some as routers (same coordinator pcb was used as a router by burining the router firmware), the router worked but it was not strong enough at all, and kept disconnected from zigbee network.

There were no options in the z2mqtt other than the name and manufacturer ... etc

Any ideas? Want the best results by using routers and coordinator.

Thank you.

**Comment by Koenkk** [comment, marked as answer]
The default is 9 ([code](https://github.com/Koenkk/Z-Stack-firmware/blob/ab4d8deb8a5203c3fbc7d292ba95083abc3f0bc5/router/Z-Stack_3.x.0/firmware.patch#L358))

**Reply by hesham1216** [reply]
Thank you very much for your reply 🙏

Can it be modified?

I found some gentleman modified the original firmwares of router and coordinator making them 20 dbm capable, based on your firmwares (of CC2652P7) 

[https://github.com/Koenkk/Z-Stack-firmware/discussions/505#discussioncomment-10611025](url)

He said he tested it, I will burn it to the module and test.

I have designed my own coordinator module and router PCB, I am testing them on a factory (big space, more devices, harsh environment), I will try to get the best result of a zigbee network following all good instructions from you guys :)

I will share my results and experience with the community after all experiments done 👍

**Reply by lux73** [reply]
As previously explained by @Koenkk: just because the coordinator shouts louder, the individual devices will still no longer reach the coordinator on the way back

best way: add a way more Router - so all devices have a stable Mesh to talk back

**Reply by hesham1216** [reply]
> As previously explained by @Koenkk: just because the coordinator shouts louder, the individual devices will still no longer reach the coordinator on the way back
> 
> best way: add a way more Router - so all devices have a stable Mesh to talk back

But after analyzing the zigbee system for some time, I noticed when the light switch for example does not receive the order, the red message appears in the Zigbee2mqtt, and it take some time to turn on or off the light, so this mean the problem is not in receiving the message back from the end device to coordinator, it is the opposite. (If the problem was in sending the feedback from the end device to the coordinator, the switch will do the order but the state in the system wouldn't change until the coordinator receive the message from end device).

One other thing I noticed after analyzing the network is the path of from routers - devices - coordinator affects the performance very very much. In the network I have more than 50+ routers.

It seems sometimes having so many routers are not healthy for the network, especially if you don't control the paths.
